src/main/cpp/option_processor.cc

```diff
@@ -305,14 +315,14 @@ blaze_exit_code::ExitCode OptionProcessor::GetRcFiles(
 
   std::vector<std::string> rc_files;
 
-  // Get the system rc (unless --nosystem_rc).
+  const std::string system_rc =
+      blaze_util::MakeAbsoluteAndResolveWindowsEnvvars(system_bazelrc_path_);
+  // Read the system rc (unless --nosystem_rc).
   if (SearchNullaryOption(cmd_line->startup_args, "system_rc", true)) {
     // MakeAbsoluteAndResolveWindowsEnvvars will standardize the form of the
     // provided path. This also means we accept relative paths, which is
     // is convenient for testing.
-    const std::string system_rc =
-        blaze_util::MakeAbsoluteAndResolveWindowsEnvvars(system_bazelrc_path_);
-    rc_files.push_back(system_rc);
+    internal::PushRcFileMaybe(&rc_files, system_rc);
```

Comment:
This will silently ignore a file passed in with `--system_rc`. That sounds dangerous.

Reply:
Only if the path is invalid or does not exist. Why does that sound dangerous?

Reply:
Because it makes typos silently do the wrong thing: if I call `bazel --system_rc=baezl.rc` it's confusing if Bazel silently ignores that file. Better to report the error.

Reply:
That's true! Do you think a fatal error is warranted, or is a warning enough?

Reply:
We error out at HEAD, don't we?

Reply:
From the code it looks like Bazel does error out if it fails to read the file:
https://github.com/bazelbuild/bazel/blob/c10468878027305d9b1ccc1a9dfbde956b788dc6/src/main/cpp/rc_file.cc#L58-L62

and that bubbles up to `blaze.cc` and the code calls `BAZEL_DIE`:

https://github.com/bazelbuild/bazel/blob/c10468878027305d9b1ccc1a9dfbde956b788dc6/src/main/cpp/blaze.cc#L1321-L1328

However tests are broken because of my changes. Apparently https://github.com/bazelbuild/bazel/pull/6896/commits/082eb9ba90670108e369fd781ed258ba6bd7cc78 broke something.